src/Results/DTO/GenerativeAiResult.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Results\DTO;
+
+use WordPress\AiClient\Files\Contracts\FileInterface;
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\Enums\MessagePartTypeEnum;
+use WordPress\AiClient\Results\Contracts\ResultInterface;
+
+/**
+ * Represents the result of a generative AI operation.
+ *
+ * This DTO contains the generated candidates along with usage statistics
+ * and metadata from the AI provider.
+ *
+ * @since n.e.x.t
+ */
+class GenerativeAiResult implements ResultInterface
+{
+    /**
+     * @var string Unique identifier for this result.
+     */
+    private string $id;
+
+    /**
+     * @var Candidate[] The generated candidates.
+     */
+    private array $candidates;
+
+    /**
+     * @var TokenUsage Token usage statistics.
+     */
+    private TokenUsage $tokenUsage;
+
+    /**
+     * @var array<string, mixed> Provider-specific metadata.
+     */
+    private array $providerMetadata;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $id Unique identifier for this result.
+     * @param Candidate[] $candidates The generated candidates.
+     * @param TokenUsage $tokenUsage Token usage statistics.
+     * @param array<string, mixed> $providerMetadata Provider-specific metadata.
+     */
+    public function __construct(string $id, array $candidates, TokenUsage $tokenUsage, array $providerMetadata = [])
+    {
+        $this->id = $id;
+        $this->candidates = $candidates;
+        $this->tokenUsage = $tokenUsage;
+        $this->providerMetadata = $providerMetadata;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Gets the generated candidates.
+     *
+     * @since n.e.x.t
+     *
+     * @return Candidate[] The candidates.
+     */
+    public function getCandidates(): array
+    {
+        return $this->candidates;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getTokenUsage(): TokenUsage
+    {
+        return $this->tokenUsage;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getProviderMetadata(): array
+    {
+        return $this->providerMetadata;
+    }
+
+    /**
+     * Converts the first candidate to text.
+     *
+     * @since n.e.x.t
+     *
+     * @return string The text content.
+     * @throws \RuntimeException If no candidates or no text content.
+     */
+    public function toText(): string
+    {
+        if (empty($this->candidates)) {
+            throw new \RuntimeException('No candidates available');
+        }
+
+        $message = $this->candidates[0]->getMessage();
+        foreach ($message->getParts() as $part) {
+            if ($part->getType()->equals(MessagePartTypeEnum::text()) && $part->getText() !== null) {
```

Comment:
Technically, the first check here is unnecessary. The type is always inferred from what kind of data is set in the part, so no need to check for that too.